geode-protobuf/src/main/java/org/apache/geode/internal/protocol/protobuf/v1/state/ProtobufConnectionAuthorizingStateProcessor.java

```diff
@@ -49,7 +50,8 @@ public void validateOperation(Object message, ProtobufSerializationService seria
     } catch (NotAuthorizedException e) {
       messageContext.getStatistics().incAuthorizationViolations();
       throw new OperationNotAuthorizedException(
-          "The user is not authorized to complete this operation");
+          "The user is not authorized to complete this operation: "
+              + ((ClientProtocol.Message) message).getMessageTypeCase());
```

Comment:
This seems to imply that the issue is specifically the type of operation being performed, but the problem may be specific to the region or key being operated on.